src/infill/SierpinskiFill.cpp

```diff
@@ -0,0 +1,802 @@
+                change = true;
+            }
+        }
+    return change;
+}
+
+bool SierpinskiFill::bubbleUpConstraintErrors()
+{
+    std::vector<std::list<std::list<SierpinskiTriangle*>::iterator>> depth_ordered(max_depth + 1);
+    { // compute depth_ordered
+        for (int i = 0; i < max_depth + 1; i++)
+        {
+            depth_ordered.emplace_back();
+        }
+        for (std::list<SierpinskiTriangle*>::iterator it = sequence.begin(); it != sequence.end(); ++it)
+        {
+            SierpinskiTriangle* node = *it;
+            depth_ordered[node->depth].emplace_back(it);
+        }
+    }
+    
+    bool redistributed_anything = false;
+    
+    for (int depth = max_depth; depth >= 0; depth--)
+    {
+        std::list<std::list<SierpinskiTriangle*>::iterator>& depth_nodes = depth_ordered[depth];
+        for (std::list<SierpinskiTriangle*>::iterator it : depth_nodes)
+        {
+            SierpinskiTriangle* node = *it;
+            SierpinskiTriangle& triangle = *node;
+            
+            float unresolvable_error = triangle.getValueError(); // node.getSubdivisionError();
+            
+            bool is_constrained_forward = isConstrainedForward(it);
+            bool is_constrained_backward = isConstrainedBackward(it);
+            if (
+                unresolvable_error > allowed_length_error &&
+                (is_constrained_forward || is_constrained_backward)
+                )
+            {
+                if (is_constrained_forward && is_constrained_backward)
+                {
+                    // when constrained in both directions, then divide error equally
+                    unresolvable_error *= .5;
+                    // disperse half of the error forward and the other half backward
+                }
+                if (deep_debug_checking) debugCheck();
+                if (is_constrained_forward)
+                {
+                    SierpinskiTriangle* next = *std::next(it);
+                    node->error_right -= unresolvable_error;
+                    next->error_left += unresolvable_error;
+                }
+                if (is_constrained_backward)
+                {
+                    SierpinskiTriangle* prev = *std::prev(it);
+                    node->error_left -= unresolvable_error;
+                    prev->error_right += unresolvable_error;
+                }
+                if (std::abs(unresolvable_error) > allowed_length_error)
+                {
+                    redistributed_anything = true;
+                }
+                if (deep_debug_checking) debugCheck();
+            }
+        }
+    }
+    return redistributed_anything;
+}
+
+
+std::list<SierpinskiFill::SierpinskiTriangle*>::iterator SierpinskiFill::subdivide(std::list<SierpinskiTriangle*>::iterator begin, std::list<SierpinskiTriangle*>::iterator end, bool redistribute_errors)
+{
+    if (redistribute_errors && deep_debug_checking) debugCheck();
```

Comment:
That line occurs in multiple places in this file. I'd rather keep the line the same everywhere where it occurs rather than changing one of them to make the code ever so slightly more compact.